src/main/java/hudson/plugins/jira/versionparameter/JiraVersionParameterDefinition.java

```diff
@@ -33,12 +35,14 @@ public JiraVersionParameterDefinition(
             String jiraProjectKey,
             String jiraReleasePattern,
             String jiraShowReleased,
-            String jiraShowArchived) {
+            String jiraShowArchived,
+            String jiraShowOnlyReleased) {
```

Comment:
Implement and align the same logic as in Jira, i.e.:
- released
- unreleased
- archived.

That will fullfil the issue in question and the logic should be simpler.

<img width="383" alt="Screenshot 2025-06-09 at 12 01 33 AM" src="https://github.com/user-attachments/assets/37b6af86-2767-4fc2-9775-2c94d0733a3f" />